addons/mail/static/tests/discuss_app/sidebar.test.js

```diff
@@ -84,6 +84,33 @@ test("toggling category button does not hide active sub thread", async () => {
     await contains(".o-mail-DiscussSidebar-item", { text: "Sub Channel" });
 });
 
+test("sub threads are sorted with last_interest_dt", async () => {
+    const pyEnv = await startServer();
+    const mainChannelId = pyEnv["discuss.channel"].create({ name: "Main Channel" });
+    const [subChannelId1] = pyEnv["discuss.channel"].create([
+        {
+            name: "Sub Channel_1",
+            parent_channel_id: mainChannelId,
+            last_interest_dt: "2021-01-02 12:00:00",
+        },
+        {
+            name: "Sub Channel_2",
+            parent_channel_id: mainChannelId,
+            last_interest_dt: "2021-01-01 12:00:00",
+        },
+    ]);
```

Comment:
Should mock date in test so this is after these dates.
I haven't checked recently if that's still the case, but in the past the HOOT tests run with year 2019, so this didn't really cover the actual test of ordering.

Also I would have at least 3 items: 2 is not enough to give confidence of proper ordering. Code could be just arbitrarily inverting the order of id.

If you have at least 3 threads A, B, C, you can check order B, C, A which gives more confidence since it shows it's not based on id in ASC or DESC order.